pkg/cmd/repo/rename/rename.go

```diff
@@ -0,0 +1,165 @@
+	IO              *iostreams.IOStreams
+	Config          func() (config.Config, error)
+	BaseRepo        func() (ghrepo.Interface, error)
+	Remotes         func() (context.Remotes, error)
+	DoConfirm       bool
+	HasRepoOverride bool
+	newRepoSelector string
+}
+
+func NewCmdRename(f *cmdutil.Factory, runf func(*RenameOptions) error) *cobra.Command {
+	opts := &RenameOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Remotes:    f.Remotes,
+		Config:     f.Config,
+	}
+
+	var confirm bool
+
+	cmd := &cobra.Command{
+		Use:   "rename [<new-name>]",
+		Short: "Rename a repository",
+		Long: heredoc.Doc(`Rename a GitHub repository
+
+		By default, renames the current repository otherwise renames the specified repository.`),
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.BaseRepo = f.BaseRepo
+			opts.HasRepoOverride = cmd.Flags().Changed("repo")
+
+			if len(args) > 0 {
+				opts.newRepoSelector = args[0]
+			} else if !opts.IO.CanPrompt() {
+				return cmdutil.FlagErrorf("new name argument required when not running interactively\n")
+			}
+
+			if len(args) == 1 && !confirm && !opts.HasRepoOverride {
+				if !opts.IO.CanPrompt() {
+					return cmdutil.FlagErrorf("--confirm required when passing a single argument")
+				}
+				opts.DoConfirm = true
+			}
+
+			if runf != nil {
+				return runf(opts)
+			}
+			return renameRun(opts)
+		},
+	}
+
+	cmdutil.EnableRepoOverride(cmd, f)
+	cmd.Flags().BoolVarP(&confirm, "confirm", "y", false, "skip confirmation prompt")
+
+	return cmd
+}
+
+func renameRun(opts *RenameOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	newRepoName := opts.newRepoSelector
+
+	currRepo, err := opts.BaseRepo()
+	if err != nil {
+		return err
+	}
+
+	if newRepoName == "" {
+		err = prompt.SurveyAskOne(
+			&survey.Input{
+				Message: fmt.Sprintf("Rename %s to: ", ghrepo.FullName(currRepo)),
+			},
+			&newRepoName,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	if opts.DoConfirm {
+		var confirmed bool
+		p := &survey.Confirm{
+			Message: fmt.Sprintf("Rename %s to %s?", ghrepo.FullName(currRepo), newRepoName),
+			Default: false,
+		}
+		err = prompt.SurveyAskOne(p, &confirmed)
+		if err != nil {
+			return fmt.Errorf("failed to prompt: %w", err)
+		}
+		if !confirmed {
+			return nil
+		}
+	}
+
+	err = runRename(httpClient, currRepo, newRepoName)
+	if err != nil {
+		return fmt.Errorf("API called failed: %s", err)
+	}
+
+	cs := opts.IO.ColorScheme()
+	newRepo := ghrepo.NewWithHost(currRepo.RepoOwner(), newRepoName, currRepo.RepoHost())
+
+	if opts.IO.IsStdoutTTY() {
+		fmt.Fprintf(opts.IO.Out, "%s Renamed repository %s\n", cs.SuccessIcon(), ghrepo.FullName(newRepo))
+	}
+
+	if !opts.HasRepoOverride {
+		remote, err := updateRemote(currRepo, newRepo, opts)
+		if err != nil {
+			fmt.Fprintf(opts.IO.ErrOut, "%s Warning: unable to update remote %q\n", cs.WarningIcon(), remote.Name)
```

Comment:
This should print the original error instead of silencing it
```suggestion
			fmt.Fprintf(opts.IO.ErrOut, "%s Warning: unable to update remote %q: %v\n", cs.WarningIcon(), remote.Name, err)
```